fibonacci/elixir/bench.exs

```diff
@@ -9,8 +9,9 @@ defmodule Bench do
   defp sum_fibs(i, max, total) when i == max, do: total
   defp sum_fibs(i, max, total), do: sum_fibs(i + 1, max, total + fib(i))
 
-  defp fib(n) when n <= 1, do: n
-  defp fib(n), do: fib(n - 1) + fib(n - 2)
+  defp fib(n, current \\ 0, next \\ 1)
+  defp fib(n, current, _next) when n == 0, do: current
+  defp fib(n, current, next), do: fib(n - 1, next, current + next)
```

Comment:
This is an iterative solution. No other implementations in this repo use an iterative solution. You'd have to ask @bddicken to be sure, but I'm pretty darn sure that the intention of the benchmark is to do it the inefficient, recursive way.

Reply:
I would agree that we need the opinion of the repo maintainer for this

But so far I don't see the problem, as per definition, fib benchmark case `Emphasizes function call overhead and recursion`, which is exactly what we're doing here

Reply:
No it isn't. The entire point of tail-call optimization is that it _doesn't_ have function call overhead. The compiler optimizes away the recursion and turns it into a loop. There is no recursion happening there after compilation.

Reply:
Thanks for your contribution!

We should only include solutions that work the "same" as the others. If we allowed this, then we'd have to open up all the other version for further optimization with tail recursion and/or iterative solutions.

Thus, as it stands now, this is disqualified. If you want to implement it in a way that matches the other solutions, send me a follow up.

Reply:
Alright, makes sense, I'll put back the other solutions back shortly